File: Semestrul 4/MPP/Proiecte/java_gui/src/main/java/map/service/DonatieService.java
package map.service;

import map.domain.Donatie;
import map.repository.database.DonatieDBRepo;
import map.repository.database.DonatieRepo0;

public class DonatieService {

    private DonatieRepo0 donatieDBRepo;

    public DonatieService(DonatieDBRepo donatieDBRepo) {
        this.donatieDBRepo = donatieDBRepo;
    }

    public void saveDonatie(Donatie donatie) {
        donatieDBRepo.save(donatie);

    }

    public int getSumaDonatiiPentruCaz(Integer id) {
        return donatieDBRepo.getSumaDonatiiPentruCaz(id);
    }
}

File: Semestrul 4/MPP/Proiecte/c#_gui/service/DonatieService.cs
using csharp_avalonia.domain;
using csharp_avalonia.repository;

namespace csharp_avalonia.service;

public class DonatieService
    {
        private readonly DonatieRepo0 donatieRepo;

        public DonatieService(DonatieRepo0 donatieRepo1)
        {
            donatieRepo = donatieRepo1;
        }

        public void SaveDonatie(Donatie donatie)
        {
            donatieRepo.save(donatie);
        }

        public int GetSumaDonatiiPentruCaz(int cazId)
        {
            return donatieRepo.GetSumaDonatiiPentruCaz(cazId);
        }
    }

